public/app/plugins/datasource/grafana/datasource.ts

```diff
@@ -58,6 +58,11 @@ class GrafanaDatasource {
       if (!_.isArray(options.annotation.tags) || options.annotation.tags.length === 0) {
         return this.$q.when([]);
       }
+      const tags = [];
+      for (let t of params.tags) {
+        tags.push(this.templateSrv.replace(t));
```

Comment:
I think it should not work in case of multi-value of the tag https://github.com/grafana/grafana/pull/9864/files#diff-e9236746ea87ffde85fa28cf23d04bf6R324

Because of the default format of the value, it will return `a,b,c` in case of many values of the tag. Then we need to split + merge